### PR TITLE
Drawing screen

### PR DESCRIPTION
**Pull request by danbarragan**
We now have a drawing screen that will render the strokes drawn on the canvas as a PNG data URL and send it over to the server via the `imageData ` param.

Also we now have client unit tests.

@mick-warehime @apairofcleats

### CODE REVIEW

File: client/src/components/DrawingScreen/DrawingScreen.test.jsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import { mount, shallow } from 'enzyme';
+import axios from 'axios';
+import DrawingScreen, { Point, Stroke } from './DrawingScreen';
+import '../../test/setupTests';
+
+describe('DrawingScreen', () => {
+  const mockGameState = { groupName: 'kitties4Life', currentPlayer: { name: 'baby cat' } };
+  const mockOnGameStateChanged = jest.fn();
+
+  it('renders strokes as a data URL', () => {
```
Comment (mick-warehime) [inline client/src/components/DrawingScreen/DrawingScreen.test.jsx:12]: why strokes as url? is that a common pattern? 

i dont have any alternative suggestions - just curious

File: client/src/components/DrawingScreen/DrawingScreen.test.jsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
...AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA8GJFFQABGYPuoAAAAABJRU5ErkJggg==';
```
Comment (mick-warehime) [inline client/src/components/DrawingScreen/DrawingScreen.test.jsx:13]: this is a bit 'unfortunate'

File: client/src/components/DrawingScreen/DrawingScreen.test.jsx [hunk 3]
```diff
@@ -0,0 +1,38 @@
+// Link.react.test.js
```
Comment (mick-warehime) [inline client/src/components/DrawingScreen/DrawingScreen.test.jsx:1]: woot. frontend tests!

File: client/src/components/DrawingScreen/DrawingScreen.test.jsx (hunk 1 above)
Comment (danbarragan) [reply client/src/components/DrawingScreen/DrawingScreen.test.jsx:12]: Well I think we have 2 options:
- Store strokes on the server (gets VERY chunky with complex drawings)
- Store images on the server (more or less same chunkdom)

Canvas elements have a method to render the contents as an image in the format of a data url that you could set the arc of an image to so I thought that'd be a good candidate for transmitting the rendered image

File: client/src/components/DrawingScreen/DrawingScreen.test.jsx (hunk 2 above)
Comment (danbarragan) [reply client/src/components/DrawingScreen/DrawingScreen.test.jsx:13]: Hahahahahaha yeaaah we could move it to a separate file and just load it from here to keep things clean but I thought meh it's just one nasty line lol. Any suggestions?

**Pull request merged**